File: LeetCode Solutions/Top Interview Questions(Easy Collection)/Array/Move Zeroes/Move Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i = 0;
        for (int j=0; j<nums.length; j++){
            if (nums[j] != 0){
                nums[i] = nums[j];
                if (j != i){
                    nums[j] = 0;
                }
                i++;
            }
        }
    }
}

File: LeetCode Solutions/Top Interview Questions(Easy Collection)/Array/Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        i = 0
        for j in range(len(nums)):
            if(nums[j] != 0):
                nums[i] = nums[j]
                if(j != i):
                    nums[j] = 0
                i += 1